test/flows/student_finance_calculator_flow_test.rb

```diff
@@ -0,0 +1,519 @@
+    setup do
+      testing_node :do_any_of_the_following_apply_uk_full_time_students_only?
+      add_responses when_does_your_course_start?: "2020-2021",
+                    what_type_of_student_are_you?: "uk-full-time",
+                    how_much_are_your_tuition_fees_per_year?: "9250",
+                    where_will_you_live_while_studying?: "at-home",
+                    whats_your_household_income?: "50,000"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node what_course_are_you_studying? for any response" do
+        assert_next_node :what_course_are_you_studying?, for_response: "children-under-17"
+      end
+    end
+  end
+
+  context "question: do_any_of_the_following_apply_all_uk_students?" do
+    setup do
+      testing_node :do_any_of_the_following_apply_all_uk_students?
+      add_responses when_does_your_course_start?: "2020-2021",
+                    what_type_of_student_are_you?: "uk-part-time",
+                    how_much_are_your_tuition_fees_per_year?: "6935",
+                    where_will_you_live_while_studying?: "at-home",
+                    whats_your_household_income?: "50,000",
+                    how_many_credits_will_you_study?: "10",
+                    how_many_credits_does_a_full_time_student_study?: "11"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of what_course_are_you_studying? for any response" do
+        assert_next_node :what_course_are_you_studying?, for_response: "has-disability"
+      end
+    end
+  end
+
+  context "question: what_course_are_you_studying?," do
+    setup do
+      testing_node :what_course_are_you_studying?
+      add_responses when_does_your_course_start?: "2020-2021",
+                    what_type_of_student_are_you?: "uk-full-time",
+                    how_much_are_your_tuition_fees_per_year?: "9250",
+                    where_will_you_live_while_studying?: "at-home",
+                    whats_your_household_income?: "50,000",
+                    do_any_of_the_following_apply_uk_full_time_students_only?: "children-under-17"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of outcome_uk_full_time_students for any of the non-medical responses" do
+        assert_next_node :outcome_uk_full_time_students, for_response: "teacher-training"
+      end
+
+      should "have a next node of are_you_a_doctor_or_dentist? for the medical response" do
+        assert_next_node :are_you_a_doctor_or_dentist?, for_response: "dental-medical-healthcare"
+      end
+
+      should "have a next node of outcome_uk_part_time_students for any response" do
+        add_responses what_type_of_student_are_you?: "uk-part-time",
```

Comment:
Don't need all the responses, just "uk-part time"

Reply:
Think they are needed here 😬

Reply:
I was wrong, definitely need more than just "uk-part-time" but you can get rid of the ones that are the same in the setup so where-will-you-live and whats-your-household-income

Reply:
in fact, if you change the setup to have `how_much_are_your_tuition_fees_per_year?: "6935",` you can remove that key from the 2nd setup too